tests/adapter/dbt/tests/adapter/grants/test_invalid_grants.py

```diff
@@ -31,29 +31,34 @@
 
 
 class BaseInvalidGrants(BaseGrants):
+    # The purpose of this test is to understand the user experience when providing
+    # an invalid 'grants' configuration. dbt will *not* try to intercept or interpret
+    # the database's own error at runtime -- it will just return those error messages.
+    # Hopefully they're helpful!
+
     @pytest.fixture(scope="class")
     def models(self):
         return {
             "my_invalid_model.sql": my_invalid_model_sql,
         }
 
-    # adapters will need to reimplement these methods with the specific
+    # Adapters will need to reimplement these methods with the specific
     # language of their database
     def grantee_does_not_exist_error(self):
         return "does not exist"
 
     def privilege_does_not_exist_error(self):
         return "unrecognized privilege"
```

Comment:
@dataders More [methods to implement](https://docs.getdbt.com/docs/contributing/building-a-new-adapter).

Reply:
Yup, there's a comment to this effect above:
https://github.com/dbt-labs/dbt-core/blob/debc867490eb724e0e7e56e68eaab7ca16168ec9/tests/adapter/dbt/tests/adapter/grants/test_invalid_grants.py#L34-L37

Is this a useful test? It's checking for specific language in the `DatabaseError` that dbt will be passing back. That may be annoying, as an adapter maintainer — this test is basically guaranteed to fail, until/unless you reimplement `grantee_does_not_exist_error` + `privilege_does_not_exist_error` with language from your own database's error message. But I do like us, as dbt maintainers, thinking hard about the UX for these common error cases.